libs/blocks/global-navigation/blocks/navDropdown/dropdown.js

```diff
@@ -0,0 +1,179 @@
+import { getAnalyticsValue, toFragment, decorateCta } from '../../utilities/utilities.js';
+import { localizeLink } from '../../../../utils/utils.js';
+
+const decorateHeadline = (elem) => {
+  if (!(elem instanceof HTMLElement)) return null;
+
+  // TODO: proper handling of aria-expanded across devices
+  const headline = toFragment`<div
+    class="feds-popup-headline"
+    role="heading"
+    aria-level="2"
+    aria-haspopup="true"
+    aria-expanded="false">
+    ${elem.textContent.trim()}
+  </div>`;
+
+  // TODO: move proper logic to accessibility,
+  // this is just for demo functionality purposes
+  headline.addEventListener('click', (e) => {
+    e.preventDefault();
+
+    const openPopup = document.querySelector('.feds-popup-headline[aria-expanded = "true"]');
+
+    if (openPopup && openPopup !== headline) {
+      openPopup.setAttribute('aria-expanded', 'false');
+    }
+
+    const currentState = headline.getAttribute('aria-expanded');
+
+    if (currentState === 'false') {
+      headline.setAttribute('aria-expanded', 'true');
+    } else {
+      headline.setAttribute('aria-expanded', 'false');
+    }
+  });
+
+  // Since heading is turned into a div, it can be safely removed
+  elem.remove();
+
+  return headline;
+};
```

Comment:
As discussed this should rather live in the keyboard navigation 👍

Reply:
Just the function from the event listener should belong to the keyboard navigation, so lines 19-35. The rest should still remain here.